client/config/cmd.go

```diff
@@ -0,0 +1,105 @@
+package config
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"path/filepath"
+
+	tmcli "github.com/tendermint/tendermint/libs/cli"
+
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+)
+
+// Cmd returns a CLI command to interactively create an application CLI
+// config file.
+func Cmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "config <key> [value]",
+		Short: "Create or query an application CLI configuration file",
+		RunE:  runConfigCmd,
+		Args:  cobra.RangeArgs(0, 2),
+	}
+	return cmd
+}
+
+func runConfigCmd(cmd *cobra.Command, args []string) error {
+
+	clientCtx := client.GetClientContextFromCmd(cmd)
+	configPath := filepath.Join(clientCtx.HomeDir, "config")
+
+	conf, err := getClientConfig(configPath, clientCtx.Viper)
+	if err != nil {
+		return fmt.Errorf("couldn't get client config: %v", err)
+	}
+
+	switch len(args) {
+	case 0:
+		// print all client config fields to sdt out
+		s, _ := json.MarshalIndent(conf, "", "\t")
+		cmd.Println(string(s))
+
+	case 1:
+		// it's a get
+
+		key := args[0]
+		switch key {
+		case flags.FlagChainID:
+			cmd.Println(conf.ChainID)
+		case flags.FlagKeyringBackend:
+			cmd.Println(conf.KeyringBackend)
+		case tmcli.OutputFlag:
+			cmd.Println(conf.Output)
+		case flags.FlagNode:
+			cmd.Println(conf.Node)
+		case flags.FlagBroadcastMode:
+			cmd.Println(conf.BroadcastMode)
+		default:
+			err := errUnknownConfigKey(key)
+			return fmt.Errorf("couldn't get the value for the key: %v, error:  %v", key, err)
+		}
+
+	case 2:
+		// it's set
+
+		key, value := args[0], args[1]
+
+		switch key {
+		case flags.FlagChainID:
+			conf.SetChainID(value)
+		case flags.FlagKeyringBackend:
+			conf.SetKeyringBackend(value)
+		case tmcli.OutputFlag:
+			conf.SetOutput(value)
+		case flags.FlagNode:
+			conf.SetNode(value)
+		case flags.FlagBroadcastMode:
+			conf.SetBroadcastMode(value)
+		default:
+			return errUnknownConfigKey(key)
+		}
+
+		configTemplate, err := initConfigTemplate()
+		if err != nil {
+			return fmt.Errorf("could not initiate config template: %v", err)
+		}
+
+		confFile := filepath.Join(configPath, "client.toml")
+		if err := writeConfigFile(confFile, conf, configTemplate); err != nil {
+			return fmt.Errorf("could not write client config to the file: %v", err)
+		}
+
+	default:
+		// print error
+		return errors.New("cound not execute config command")
```

Comment:
This should be a `panic()` as this line should never be executed (cobra handles argument validation)

Reply:
I heard, we try to avoid using `panic()' in our code. I would suggest to remove default case from switch construction. Does it sound good?

Reply:
I can't quite get this `panic`-phobia. `panic` is meant to be used when something entirely unpredictable and unexpected causes an _unrecoverable_ error. From what I see, line 97 takes a divine intervention to be executed. This is truly the best example of when to use `panic`

Reply:
I just tried  `./build/simd config node 1  2` Output: `Error: accepts between 0 and 2 arg(s), received 3`.

Reply:
Yes. It's still good coding practice to handle the unpredictable by panicking and spitting out as much as debugging information as possible. Which is what panic() does.

Reply:
Done. Thanks for suggestion.